keps/sig-node/compute-device-assignment.md

```diff
@@ -0,0 +1,187 @@
+---
+kep-number: 18
+title: Container API and Kubelet extension for device assignment details 
+authors:
+  - "@dashpole" 
+  - "@vikaschoudhary16"
+owning-sig: sig-node
+reviewers:
+  - "@thockin"
+  - "@derekwaynecarr"
+  - "@dchen1107"
+  - "@vishh"
+approvers:
+  - "@sig-node-leads"
+editors:
+  - "@dashpole"
+  - "@vikaschoudhary16"
+creation-date: "2018-07-19"
+last-updated: "2018-07-19"
+status: provisional
+---
+# Container API and Kubelet extension for device assignment details 
+
+Table of Contents
+=================
+* [Abstract](#abstract)
+* [Background](#background)
+* [Objectives](#objectives)
+* [Non Objectives](#non-objectives)
+* [User Journeys](#user-journeys)
+  * [Device Monitoring Agents](#device-monitoring-agents)
+  * [Out-of-tree Experimentation](#out-of-tree-experimentation)
+    * [Network Device Plugins](#network-device-plugins)
+    * [Guaranteed Device Scheduling](#guaranteed-device-scheduling)
+* [Alternatives Considered](#alternatives-considered)
+* [Changes](#changes)
+  * [API Changes](#api-changes)
+  * [Kubelet Changes](#kubelet-changes)
+    * [Pod Admission and Device Allocation](pod-admission-device-allocation)
+    * [Updating ComputeDevices in the pod spec at api server](updating-compute-devices)
+
+## Abstract
+In this document we will discuss the motivation and code changes required for introducing assigned device IDs information in the container API object.
+
+## Background
+The [Pod API Object](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L3072) contains the metadata, specification, and status of the smallest unit of computing that can be managed by kubernetes. In order to support Network Devices through the device plugin, it has been proposed (in [Network Devices Support Changes Proposal](https://docs.google.com/document/d/1TpYqgmJQOctm1oexQRo0cGNb9oyk0vozhmhr6myY99M/edit#heading=h.dp2sl8erwljy)) to add Pod UID to the `Allocate` call, and then have the CNI ask the Network Device Plugin for it to do network plumbing. This proposal takes a different approach that does not rely on adding the `PodUID` to the `Allocate` DPAPI call, and does not require explicit coordination between CNI and the Device Plugin.
+
+[Device Monitoring](https://docs.google.com/document/d/1NYnqw-HDQ6Y3L_mk85Q3wkxDtGNWTxpsedsgw4NgWpg/edit?usp=sharing) requires external agents to be able to determine the set of devices in-use by containers and attach pod and container metadata for these devices.
+
+## Objectives
+
+* To enable node monitoring agents to determine the set of devices that are in-use, and to obtain metadata about the container/pod using them.
```

Comment:
is eliminating kubelets knowledge of gpus directly an objective that we can use to validate the design?  for example, should this result in removing `AcceleratorStats` from the summary stats api?

Reply:
Ideally, the summary stats API should by itself by refactored into core and non-core metrics. When that happens, we can drop accelerator metrics from the core metrics APIs. 
Coincidentally, @dashpole is the expert on summary APIs as well :)